Lib/test/test_zipfile/test_core.py

```diff
@@ -1780,6 +1780,34 @@ def test_writestr_extended_local_header_issue1202(self):
                 zinfo.flag_bits |= zipfile._MASK_USE_DATA_DESCRIPTOR  # Include an extended local header.
                 orig_zip.writestr(zinfo, data)
 
+    def test_write_with_source_date_epoch(self):
+        # Set the SOURCE_DATE_EPOCH environment variable to a specific timestamp
+        os.environ['SOURCE_DATE_EPOCH'] = "1727440508"
+
+        with zipfile.ZipFile(TESTFN, "w") as zf:
+            zf.writestr("test_source_date_epoch.txt", "Testing SOURCE_DATE_EPOCH")
+
+        with zipfile.ZipFile(TESTFN, "r") as zf:
+            zip_info = zf.getinfo("test_source_date_epoch.txt")
+            get_time = time.gmtime(int(os.environ['SOURCE_DATE_EPOCH']))[:6]
+            # Compare each element of the date_time tuple
+            # Allow for a 1-second difference
+            for z_time, g_time in zip(zip_info.date_time, get_time):
```

Comment:
The `assertAlmostEqual` method is used with a `delta=1` to allow for a maximum difference of 1 second between corresponding elements. This tolerance is important because of small potential discrepancies in system time calculations, especially when writing and reading back timestamps in compressed files. The delta ensures that minor differences do not cause test failures if the times are very close (within 1 second).